### PR TITLE
Pip 248

### LINKED ISSUE #347: Check for log4shell vulnerability
https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-44228
https://logging.apache.org/log4j/2.x/security.html

### LINKED ISSUE #357: Elasticsearch and log4j vulnerability
Hi

Elasticsearch has released a updated version that fixes there internal log4j vulnerabilities. 
(https://www.elastic.co/blog/new-elasticsearch-and-logstash-releases-upgrade-apache-log4j2)

I can see that the docker files in this repo are still pointing to EL version 7.7.0
Can theese be updated to v 7.16.2 or will that break the frontend of Snowstorm?

//Per

### PR DESCRIPTION
**Pull request by jimcornmell**
Added BOM and thus security fixes
HAPI FHIR upgrade, required to lib update for security.

FIXES #347 
FIXES #357

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,60 +5,55 @@
...
 	<artifactId>snowstorm</artifactId>
 	<version>8.2.0-SNAPSHOT</version>
 	<parent>
-		<groupId>org.springframework.boot</groupId>
-		<artifactId>spring-boot-starter-parent</artifactId>
-		<version>2.7.0</version>
-		<!-- Remember to update documentation links if this version changes -->
+		<groupId>org.snomed</groupId>
+		<artifactId>snomed-parent-bom</artifactId>
+		<version>1.0.4-SNAPSHOT</version>
```
Comment (kaicode) [inline pom.xml:13]: You will need 1.0.5 or later here.

File: pom.xml [hunk 2]
```diff
@@ -5,60 +5,55 @@
...
-		<spring-cloud.version>2021.0.3</spring-cloud.version>
-
-		<!-- FHIR HAPI library -->
-		<hapi.version>5.6.3-si</hapi.version>
-
-		<docker.image.prefix>snomedinternational</docker.image.prefix>
-		<slf4j.version>1.7.36</slf4j.version>
-		<log4j.version>2.17.0</log4j.version>
-		<aws-java-sdk.version>1.11.221</aws-java-sdk.version>
+		<snomed-owl-toolkit.version>3.1.0-SNAPSHOT</snomed-owl-toolkit.version>
```
Comment (kaicode) [inline pom.xml:22]: This line is not needed. It's needed in SnowstormX because that uses an experimental owl-toolkit version.

**Review by kaicode — COMMENTED** [review]
I think there are two minor pom changes to fix, then it looks good to me.

**Review by kaicode — APPROVED** [review]
Looks good to me!

File: pom.xml
Comment (kaicode) [inline pom.xml:1]: @jimcornmell I found that I needed to add the repackage goal (`<goal>repackage</goal>`) to the spring-boot-maven-plugin to get it to create an executable snowstorm jar. This must be related to the spring-boot upgrade. I have only just discovered this.

Comment (CoderMChu) [reply pom.xml:1]: Yes I had to change for classification-service too.

**Pull request merged**